Rewrite Overseas France parser (#3461)

RobinFrcd opened the pull request:
- Currently, La Réunion is the only territory with live data
  (https://opendata-reunion.edf.fr/)
- Historical data is fetched from https://opendata-corse-outremer.edf.fr
- No price data available after 2019
- Simplified parser. @PETILLON-Sebastien did a great job but it wasn't
  compatible with current data/projet structure anymore

RobinFrcd commented:
@pierresegonne @skovhus @corradio Hi guys, sorry about the ping, but Réunion island is really broken right now. I've tried to simplify everything, is it too much for you ? Do you think we should just fix the coal issue ?

madsnedergaard commented:
> @pierresegonne @skovhus @corradio Hi guys, sorry about the ping, but Réunion island is really broken right now. I've tried to simplify everything, is it too much for you ? Do you think we should just fix the coal issue ?

Hey @RobinFrcd, can you elaborate on why it's really broken right now? Also, will #3472 fix any of the problems?

RobinFrcd commented:
Hi @madsnedergaard,
Right now, coal is broken for La Réunion. I think we used to have bagasse, coal and gas. Now we have coal, gas has been removed and bagassed has been merged into biomass. 

I just wanted to simplify the script and rely on the API data for everything. 
Also, historical data is borken:

```
poetry run test_parser RE --target_datetime 2019-01-01
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/robin/dev/python/electricitymap-contrib/.venv/lib/python3.7/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/home/robin/dev/python/electricitymap-contrib/.venv/lib/python3.7/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/home/robin/dev/python/electricitymap-contrib/.venv/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/robin/dev/python/electricitymap-contrib/.venv/lib/python3.7/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/home/robin/dev/python/electricitymap-contrib/test_parser.py", line 68, in test_parser
    *args, target_datetime=target_datetime, logger=logging.getLogger(__name__)
  File "/home/robin/dev/python/electricitymap-contrib/parsers/FR_O.py", line 222, in fetch_production
    params = get_param(zone_key, target_datetime)
  File "/home/robin/dev/python/electricitymap-contrib/parsers/FR_O.py", line 43, in get_param
    datetime = arrow.get(target_datetime, tz='Europe/Paris')
  File "/home/robin/dev/python/electricitymap-contrib/.venv/lib/python3.7/site-packages/arrow/api.py", line 21, in get
    return _factory.get(*args, **kwargs)
  File "/home/robin/dev/python/electricitymap-contrib/.venv/lib/python3.7/site-packages/arrow/factory.py", line 260, in get
    return self.type(*args, **kwargs)
TypeError: __init__() missing 2 required positional arguments: 'month' and 'day'
```

I simplified the script, and fixed that.

Yes, #3472 fixes the coal issue

madsnedergaard commented:
Hey @RobinFrcd, if you can update this PR after #3472 we'd be happy to review it and get it merged :)

VIKTORVAV99 commented:
It shouldn't introduce any major changes but since #3978 was merged FR-COR is included in the parser as well.

madsnedergaard commented:
Hey, does anyone want to continue work on this PR and get it merged? :)

VIKTORVAV99 commented:
> Hey, does anyone want to continue work on this PR and get it merged? :)

I can take a crack at rewriting it but as I mentioned to @pierresegonne on slack the API also contain estimated data before it has been validated (in the same endpoint).

We discussed it a little and one solution would be to enable the backend to consume estimated data (separate from production data). I think that is the biggest blocker for now if we don't want data that is delayed by months if not years.
